internal/section/service_test.go

```diff
@@ -182,9 +187,53 @@ func TestServiceDelete(t *testing.T) {
 		repository.On("Get", id).Return(&mockedSection)
 		repository.On("Delete", id)
 
-		err := service.Delete(context.TODO(), id)
+		err := service.Delete(id)
+
+		assert.NoError(t, err)
+	})
+}
+
+func TestCountProductsByAllSections(t *testing.T) {
+	t.Run("Should return a list of sections", func(t *testing.T) {
+		service, repository := CreateService(t)
+
+		expected := []domain.ProductsBySectionReport{mockedProductsBySection}
+
+		repository.On("CountProductsByAllSections").Return(expected)
+		result := service.CountProductsByAllSections()
+
+		assert.NotEmpty(t, result)
+		assert.True(t, len(result) == 1)
+		assert.Equal(t, result[0], mockedProductsBySection)
+	})
+}
+
+func TestCountProductsBySection(t *testing.T) {
+	t.Run("Should return  product by section by a specified id", func(t *testing.T) {
```

Comment:
Sugiro alterar a descrição para "Should return  amount of products by section by a specified id"

Reply:
Feito!